routes/kingdoms.js

```diff
@@ -0,0 +1,107 @@
+const express = require('express');
+const router = express.Router();
+const { getKingdoms, getCastles, getLieges, getVassals } = require('../services/kingdom_getters');
+const { addKingdom, addCastles, addLiegies, addVassals } = require('../services/kingdom_setters');
+const { deleteLiege, deleteCastle, deleteKingdom, deleteVassal } = require('../services/kingdom_deletes');
+
+
+router.get("/", (req, res) => {
+    const getAllKingdoms = getKingdoms();
+
+    res.render("kingdoms", { getAllKingdoms });
+});
+
+router.get('/:kingdom', (req, res) => {
+    const kingdomSelected = req.params.kingdom;
+    //get all castles for the selected kingdom
+    const castles = getCastles(kingdomSelected);
+
+    res.render('kingdom/show', { kingdomSelected, castles });
+});
+
+router.get('/:kingdom/castles/:castle', (req, res) => {
+    const kingdomSelected = req.params.kingdom;
+    const castleSelected = req.params.castle;
+    const lieges = getLieges(kingdomSelected, castleSelected);
+
+    res.render('castle/show', { kingdomSelected, castleSelected, lieges });
+
+});
+
+router.get('/:kingdom/castles/:castle/lieges/:liege/vassals', (req, res) => {
+    const kingdomSelected = req.params.kingdom;
+    const castleSelected = req.params.castle;
+    const liegeSelected = req.params.liege;
+    const vassals = getVassals(kingdomSelected, castleSelected, liegeSelected);
+
+    res.render('liege/show', { kingdomSelected, castleSelected, liegeSelected, vassals });
+
+});
+
+router.post('/', (req, res) => {
+    const kingdomName = req.body.kingdom;
+    const kingName = req.body.king;
+    const queenName = req.body.queen;
+    addKingdom(kingdomName, kingName, queenName);
```

Comment:
These are all great looking routes. All of the business logic is encapsulated within other modules. Nicely done 🎉